src/equations/numerical_fluxes.jl

```diff
@@ -0,0 +1,148 @@
+
+@inline function (dissipation::DissipationGlobalLaxFriedrichs)(u_ll, u_rr, orientation, equations)
+  @unpack λ = dissipation
+  return -λ/2 * (u_rr - u_ll)
+end
+
+Base.string(d::DissipationGlobalLaxFriedrichs) = "DissipationGlobalLaxFriedrichs(" * string(d.λ) * ")"
+
+
+"""
+    DissipationLocalLaxFriedrichs(max_abs_speed=max_abs_speed_naive)
+
+Create a local Lax-Friedrichs dissipation operator where the maximum absolute wave speed
+is estimated as `max_abs_speed(u_ll, u_rr, orientation, equations)`, defaulting to
+[`max_abs_speed_naive`](@ref).
+"""
+struct DissipationLocalLaxFriedrichs{MaxAbsSpeed}
+  max_abs_speed::MaxAbsSpeed
+end
+
+DissipationLocalLaxFriedrichs() = DissipationLocalLaxFriedrichs(max_abs_speed_naive)
+
+@inline function (dissipation::DissipationLocalLaxFriedrichs)(u_ll, u_rr, orientation, equations)
+  λ = dissipation.max_abs_speed(u_ll, u_rr, orientation, equations)
+  return -0.5 * λ * (u_rr - u_ll)
+end
+
+Base.string(d::DissipationLocalLaxFriedrichs) = "DissipationLocalLaxFriedrichs(" * string(d.max_abs_speed) * ")"
+
+
+"""
+    max_abs_speed_naive(u_ll, u_rr, orientation, equations)
+
+Simple and fast estimate of the maximal wave speed of the Riemann problem with left and right states
+`u_ll, u_rr`, based only on the local wave speeds associated to `u_ll` and `u_rr`.
+"""
+function max_abs_speed_naive end
+
+
+const FluxLaxFriedrichs{MaxAbsSpeed} = FluxPlusDissipation{typeof(flux_central), DissipationLocalLaxFriedrichs{MaxAbsSpeed}}
+"""
+    FluxLaxFriedrichs(max_abs_speed=max_abs_speed_naive)
+
+Local Lax-Friedrichs (Rusanov) flux with maximum wave speed estimate provided by
+`max_abs_speed`, cf. [`DissipationLocalLaxFriedrichs`](@ref) and
+[`max_abs_speed_naive`](@ref).
+"""
+function FluxLaxFriedrichs(max_abs_speed=max_abs_speed_naive)
+  FluxPlusDissipation(flux_central, DissipationLocalLaxFriedrichs(max_abs_speed))
+end
+
+Base.string(f::FluxLaxFriedrichs) = "FluxLaxFriedrichs(" * string(f.dissipation.max_abs_speed) * ")"
+
+# TODO: Shall we deprecate `flux_lax_friedrichs`?
+const flux_lax_friedrichs = FluxLaxFriedrichs()
+
+
+"""
+    FluxHLL(min_max_speed=min_max_speed_naive)
+
+Create an HLL (Harten, Lax, van Leer) numerical flux where the minimum and maximum
+wave speeds are estimated as `λ_min, λ_max = min_max_speed(u_ll, u_rr, orientation, equations)`,
+defaulting to [`min_max_speed_naive`](@ref).
+"""
+struct FluxHLL{MinMaxSpeed}
+  min_max_speed::MinMaxSpeed
+end
+
+FluxHLL() = FluxHLL(min_max_speed_naive)
+
+"""
+    min_max_speed_naive(u_ll, u_rr, orientation, equations)
+
+Simple and fast estimate of the minimal and maximal wave speed of the Riemann problem with
+left and right states `u_ll, u_rr`, usually based only on the local wave speeds associated to
+`u_ll` and `u_rr`.
+"""
+function min_max_speed_naive end
+
+@inline function (numflux::FluxHLL)(u_ll, u_rr, orientation, equations)
+  λ_min, λ_max = numflux.min_max_speed(u_ll, u_rr, orientation, equations)
+
+  if λ_min >= 0 && λ_max >= 0
```

Comment:
This is not the same implementation as we use currently, cf. https://github.com/trixi-framework/Trixi.jl/blob/main/src/equations/compressible_euler_2d.jl#L875 (I also cursorily checked the other equations, they all seem to use the same >= then > version). Maybe your version of the HLL flux is the correct one, but if possible I'd keep this PR as a refactoring with identical results and fix the HLL implementation in a separate PR.

Reply:
I wanted to avoid the possible division by zero if both estimates vanish. But we can also use the previous behavior and open an issue to discuss this further.

Reply:
You are probably right, I'd just like to keep this a separate PR since it potentially changes the behavior of existing simulations (unlikely, true, but we have seen so many issues around the symmetry of the HLL flux that I just don't want it to bite us again...). An issue sounds good!

Reply:
Done in 883fae99210288ceac3eab10e0f31f94b6f547b3.

Reply:
Thanks!